test/e2e/client.go

```diff
@@ -257,6 +277,167 @@ func (c *Client) WaitForObjectCreated(obj client.Object) error {
 }
 
 func (c *Client) ForwardPort(serviceName, namespaceName string, port int) (string, func(), error) {
+	return c.ForwardPortV2(serviceName, namespaceName, port)
+}
+
+// ForwardPortV2 is the new implementation using service and pod selector, select running but not terminating pods
+func (c *Client) ForwardPortV2(serviceName, namespaceName string, port int) (string, func(), error) {
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: namespaceName,
+		},
+	}
+	err := c.Get(service)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to get service %s/%s: %w", namespaceName, serviceName, err)
+	}
+
+	var targetPort int32 = 0
+	if len(service.Spec.Ports) == 0 {
+		return "", nil, fmt.Errorf("service %s/%s has no ports defined", namespaceName, serviceName)
+	}
+
+	// Find the port that matches the requested port
+	for _, svcPort := range service.Spec.Ports {
+		if svcPort.Port == int32(port) {
+			if svcPort.TargetPort.IntVal != 0 {
```

Comment:
Target port is a string   in our config @raptorsun 
----
ports:
  - name: https
    port: 8443
    protocol: TCP
    targetPort: https